keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/probabilistic/BetaVertex.java

```diff
@@ -73,46 +77,31 @@ public BetaVertex(int[] tensorShape, double alpha, double beta) {
 
     @Override
     public double logPdf(DoubleTensor value) {
-
-        DoubleTensor alphaValues = alpha.getValue();
-        DoubleTensor betaValues = beta.getValue();
-
-        DoubleTensor logPdfs = Beta.logPdf(alphaValues, betaValues, value);
-
+        DoubleTensor logPdfs = distribution().logProb(value);
         return logPdfs.sum();
     }
 
     @Override
     public Map<Long, DoubleTensor> dLogPdf(DoubleTensor value) {
-        Beta.Diff dlnP = Beta.dlnPdf(alpha.getValue(), beta.getValue(), value);
-        return convertDualNumbersToDiff(dlnP.dPdalpha, dlnP.dPdbeta, dlnP.dPdx);
+        List<DoubleTensor> dlnP = distribution().dLogProb(value);
+        return convertDualNumbersToDiff(dlnP);
     }
 
-    private Map<Long, DoubleTensor> convertDualNumbersToDiff(DoubleTensor dPdalpha,
-                                                             DoubleTensor dPdbeta,
-                                                             DoubleTensor dPdx) {
-
-        PartialDerivatives dPdInputsFromAlpha = alpha.getDualNumber().getPartialDerivatives().multiplyBy(dPdalpha);
-        PartialDerivatives dPdInputsFromBeta = beta.getDualNumber().getPartialDerivatives().multiplyBy(dPdbeta);
+    private Map<Long,DoubleTensor> convertDualNumbersToDiff(List<DoubleTensor> dlnP) {
```

Comment:
I prefer the way this looks in `GammaVertex`, `GaussianVertex` etc, can we change it to match the others?

```
private Map<Long,DoubleTensor> convertDualNumbersToDiff(DoubleTensor dPdalpha, DoubleTensor dPdbeta, DoubleTensor dPdx) {
```

And then do the get's from our list of derivatives earlier on?

Reply:
Yes absolutely. The only reason this one looks different is because I did it first, and didn't go back to change it.